minidump/src/minidump.rs

```diff
@@ -2864,7 +2864,10 @@ impl<'a> MinidumpStream<'a> for MinidumpMacCrashInfo {
         let strings_offset = header.record_start_size as usize;
         let mut prev_version = None;
         let mut infos = Vec::new();
-        for record_location in &header.records[..header.record_count as usize] {
+
+        let records = header.records.iter().take(header.record_count as usize);
```

Comment:
Can you add a comment here noting this is a concern? Something to the effect of:

```
// We use `take` here to better handle a corrupt record_count that is larger than the maximum supported size
```

Reply:
Yep, that's done!